server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesSourceParserHelper.java

```diff
@@ -67,6 +68,8 @@ public static void writeTo(CompositeValuesSourceBuilder<?> builder, StreamOutput
             code = 1;
         } else if (builder.getClass() == HistogramValuesSourceBuilder.class) {
             code = 2;
+        } else if (builder.getClass() == GeoHashGridValuesSourceBuilder.class && out.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
I am not sure the best practices for aggregation bwc streaming. 

If we serialize the code and object here, the older node would throw an `Invalid code`, if we block here, we throw an `invalid builder type`. I am not sure which is "best"

Reply:
+1 to throw the exception when serializing, I agree that we should throw early if we know that the remote end cannot handle the agg.

Reply:
Cool, I will update here and provide a nicer message.